src/astUtils/visitors.ts

```diff
@@ -0,0 +1,169 @@
+import { CancellationToken } from 'vscode-languageserver';
+import { Statement, Body, AssignmentStatement, Block, ExpressionStatement, CommentStatement, ExitForStatement, ExitWhileStatement, FunctionStatement, IfStatement, IncrementStatement, PrintStatement, GotoStatement, LabelStatement, ReturnStatement, EndStatement, StopStatement, ForStatement, ForEachStatement, WhileStatement, DottedSetStatement, IndexedSetStatement, LibraryStatement, NamespaceStatement, ImportStatement } from '../parser/Statement';
+import { Expression } from '../parser/Expression';
+import { isExpression, isBlock, isIfStatement } from './reflection';
+
+
+/**
+ * Create a filtered visitor for use with `editStatements`
+ */
+export function createStatementVisitor(
```

Comment:
Are we missing class statement here too? What about class fields/methods?

Reply:
This visitor only visits blocks - I don't think you can have a class in a Block?
Visiting classes/fields for now is meant to be visited using `file.parser.classStatements` collection.

Reply:
Ok, thanks for explaining.